### ISSUE TITLE
multi-part MIME not correctly handled

### ISSUE THREAD

**Issue by noktilux** [papnkukn/eml-format#4]
here is a message containing a small attached jpeg and a little message ("please see attached").  your parser is only picking up the attached jpeg and not getting the little message:

http://qstatistic.com/debug/sample_email.txt

i do not get the text with either the "read" or "parse" function.

**Comment by noktilux** [comment]
i have had a look at the code and the issue is in this line:

`if (lines[i - 1] == "" && line.indexOf("--" + findBoundary) == 0 && !/\-\-(\r?\n)?$/g.test(line)) {`

the first bit -- looking for empty string -- is not the valid in the test message i linked to above.

line[i-1] consists of "This is a multi-part message in MIME format."

**Comment by hi2u** [comment]
I was also having problems parsing over 50% of my emails due to some mime header content I think.    Not sure if your issue is related, but here's what worked for me...

### Didn't work:

```js
emlformat.read(eml, { headersOnly: false }, function(error, data) {...}
```

The error I was getting was:

```
TypeError: Cannot read property 'length' of undefined
    at _read (node_modules/eml-format/lib/eml-format.js:466:39)
    at node_modules/eml-format/lib/eml-format.js:518:7
    at Object.emlformat.parse (node_modules/eml-format/lib/eml-format.js:554:5)
    at Object.emlformat.read (node_modules/eml-format/lib/eml-format.js:516:15)
```

### Did work:

I just changed the `headersOnly` option to `false` and 100% of my emails were parsed...

```js
emlformat.read(eml, { headersOnly: false }, function(error, data)
```

**Comment by papnkukn** [comment]
@noktilux thanks for providing the example.

That is correct, the issue is in the condition that strictly requires **an empty line before** the multi-part boundary marker
```js
if (lines[i - 1] == "" && line.indexOf("--" + findBoundary) == 0 && !/\-\-(\r?\n)?$/g.test(line)) {
```

Solved by removing the `lines[i - 1] == ""` ("previous line should be blank") condition from the `if` statement.

Issues has been fixed with version 0.6.0.

**Comment by papnkukn** [comment]
Just to provide an example.

So if the EML looks like this, i.e. with no new line after *This is a multi-part message in MIME format.*
```
....
MIME-Version: 1.0
Content-Type: multipart/mixed;
 boundary="------------194F0B6C07FF2414138ED9B2"
Content-Language: en-US

This is a multi-part message in MIME format.
--------------194F0B6C07FF2414138ED9B2
Content-Type: text/plain; charset=utf-8; format=flowed
Content-Transfer-Encoding: 7bit

please see attached


--------------194F0B6C07FF2414138ED9B2
Content-Type: image/jpeg;
 name="tired_boot.FJ010019.jpeg"
...
```

The **eml-format** should now read it as
```js
{
  "date": "2018-04-29T18:05:09.000Z",
  ...
  "text": "please see attached\r\n\r\n",
  "attachments": [
    {
      "name": "tired_boot.FJ010019.jpeg", 
      ...
}
```
with the `text` property.
